packages/glsp-playwright-core/src/extension/capabilities/resize-handle/resize-handle.capability.ts

```diff
@@ -0,0 +1,23 @@
+import type { PNode } from '~/glsp/graph';
+import type { ConstructorA } from '~/types';
+import type { Clickable } from '../../flows';
+import type { Capability } from '../../types';
+import { ResizeHandles } from './resize-handle.po';
+
+export interface ResizeHandleCapability<TResizeHandles extends ResizeHandles = ResizeHandles> {
+    resizeHandles(): TResizeHandles;
+}
+
+export function ResizeHandleCapability<TBase extends ConstructorA<PNode & Clickable>>(
+    Base: TBase
+): Capability<TBase, ResizeHandleCapability> {
+    abstract class Mixin extends Base implements ResizeHandleCapability {
+        #handles = new ResizeHandles(this);
```

Comment:
Is there a particular reason you use `#` within the name here, seems odd to me.

Reply:
`#` Makes it "truly" private in classes and not faking it (it is relatively new)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
Oh cool, I actually never used that before. Thank you for the link!

Btw, I have another question regarding Mixin classes: Is there any way to extend them from outside? Just from a very quick glance, I couldn't see a way but if there is, we definitely should not make fields completely "hidden" private but rather protected - simply from a framework point of view. If there is no way to do that, do you think it would make sense to allow that somehow? In this case it is not very interesting but, for instance, in the `drag.flow.ts` there is actually quite some implementation in the methods.

I also noticed that sometimes you create the objects once (like here) and sometimes you create them in the method (e.g., `resizeHandles`), is there a particular reason for that? I assume it has to do with the time that the object locates the DOM elements but I want to confirm with you.

Reply:
Extending is possible by either

- Creating custom functions that implement **the interface** (everything new)
- Creating custom functions that extend the **default class** and override the methods (overriding)

Further, the `Mixins` (most of the time) use a separate `Page Object` and that can also be extended / replaced as required. 

**Summarized**: Yes, it can be changed completely (new class) or extended (extend other class and override methods). 

However, `Mixins` do not allow protected methods (TS does not support it). 

I though about it. `Mixins` should not require any `protected` fields / methods. Because `Mixins` either provide complete functionality or reuse `Page Objects` to overcome complexity (e.g., `command-palette`) and those `Page Objects` can be replaced by the developers (e.g., override them). For those `Page Objects` protected fields / methods are still possible. Consequently, the `Mixins` should only implement the **public contract / API** and provide easier access to the `Page Object` or provide small isolated functionality and not require any delegation to child classes.

> I also noticed that sometimes you create ...

=> I have to move the `Page Object` to the method.